src/charm.py

```diff
@@ -146,6 +162,23 @@ def _on_tracing_relation_changed(self, e: RelationEvent):
         if not self._tracing.is_v2(e.relation):
             self._publish_v1_data(e.relation)
 
+    def _on_ingress_relation_created(self, e: RelationEvent):
+        if self._ingress.is_ready():
```

Comment:
duplicated code between created and joined handler, consolidate?

Reply:
IIUC you could just call `_configure_ingress()` and be done with it, no?

Reply:
True - changed that.